wire-ios-system/Tests/SplitViewControllerDelegate/SupportedOrientationsDelegatingSplitViewControllerDelegateTests.swift

```diff
@@ -0,0 +1,146 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import XCTest
+
+@testable import WireSystem
+
+final class SupportedOrientationsDelegatingSplitViewControllerDelegateTests: XCTestCase {
+
+    private var sut: SupportedOrientationsDelegatingSplitViewControllerDelegate!
+
+    override func setUp() async throws {
+        sut = await MainActor.run { .init() }
+    }
+
+    override func tearDown() {
+        sut = nil
+    }
+
+    @MainActor
+    func testAllOrientationsSupported() {
+
+        // Given
+        let splitViewController = UISplitViewController(style: .doubleColumn)
+        splitViewController.setViewController(ViewController(), for: .primary)
+        splitViewController.setViewController(ViewController(), for: .secondary)
+
+        // When
+        let result = sut.splitViewControllerSupportedInterfaceOrientations(splitViewController)
+
+        // Then
+        XCTAssertEqual(result, .all)
+    }
+
+    @MainActor
+    func testOnlyUpsideDownSupported() {
+
+        // Given
+        let splitViewController = UISplitViewController(style: .tripleColumn)
+        splitViewController.setViewController(ViewController([.portrait, .portraitUpsideDown]), for: .primary)
+        splitViewController.setViewController(ViewController([.landscapeLeft, .portraitUpsideDown]), for: .supplementary)
+        splitViewController.setViewController(ViewController([.landscapeRight, .portraitUpsideDown]), for: .secondary)
+
+        // When
+        let result = sut.splitViewControllerSupportedInterfaceOrientations(splitViewController)
+
+        // Then
+        XCTAssertEqual(result, .portraitUpsideDown)
+    }
+
+    @MainActor
+    func testAllButUpsideDownSupported() {
```

Comment:
Is this test named correctly considering its behavior? I expected `result` to be all orientations except the one that it actually is.